sched/scheduler/cluster_state.go

```diff
@@ -13,16 +13,16 @@ const noTask = ""
 type clusterState struct {
 	updateCh chan []cluster.NodeUpdate
 	nodes    map[cluster.NodeId]*nodeState
-	affinity map[string]*nodeSet
+	affinity map[string]*nodeGroups //key is a snapshotID.
```

Comment:
Sorry, I wasn't clear.... I was recommending changing affinity to nodeGroups.
the nodeGroups type should be singular (nodeGroup)... 

type clusterState struct {
...
    nodeGroups map[string]*nodeGroup // key is snapshotID.
}

type nodeGroup struct {
...
}